br/pkg/task/restore.go

```diff
@@ -895,18 +901,48 @@ func RunRestore(c context.Context, g glue.Glue, cmdName string, cfg *RestoreConf
 			restoreErr = err
 			return
 		}
-		tableIds := make([]int64, 0, len(cfg.PiTRTableTracker.TableIdToDBIds))
-		for tableId := range cfg.PiTRTableTracker.TableIdToDBIds {
-			tableIds = append(tableIds, tableId)
-		}
-		for tableId := range cfg.PiTRTableTracker.PartitionIds {
-			tableIds = append(tableIds, tableId)
+
+		// Extract downstream IDs from tableMappingManager
+		// ApplyFilterToDBReplaceMap has already filtered the DBReplaceMap based on PiTRTableTracker,
+		// so we can directly iterate through it and collect non-filtered IDs
+		var downstreamTableIds []int64
```

Comment:
use `map[int64]struct{}` instead, because table id may appear repeatedly. For example, `rename table test1.t1 to test2.t1`, table id will appear in `DBReplaceMap[test1.id].TableMap[t1.id]` and `DBReplaceMap[test2.id].TableMap[t1.id]`.